### PR TITLE
Allows Borers to Create Changelings + Cleans up Borer Code + Changes Brain to Random Organ Damage

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

WIP Borer re-work by yours truly, will finish this bit after the meat is done cooking. 

## How This Contributes To The Skyrat Roleplay Experience

Should be fun! 

## Changelog

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 1]
```diff
@@ -105,11 +105,12 @@ GLOBAL_LIST_EMPTY(cortical_borers)
...
 										/datum/reagent/medicine/inacusiate,
 										/datum/reagent/medicine/oculine,
+										/datum/reagent/medicine/mannitol,
 										/datum/reagent/toxin/mindbreaker,
+										/datum/reagent/medicine/insulin,
 	)
 	//blacklisted chemicals - separate from chemicals that cannot be synthesized, borers specifically cannot learn these
-	var/list/blacklisted_chemicals = list(/datum/reagent/medicine/mannitol,
-										/datum/reagent/medicine/neurine,
+	var/list/blacklisted_chemicals = list(/datum/reagent/medicine/adminordrazine,
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:113]: redundant-- borers can only get/learn synthesizable chems.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 2]
```diff
@@ -164,11 +165,11 @@ GLOBAL_LIST_EMPTY(cortical_borers)
 	///we dont want to spam the chat
 	var/deathgasp_once = FALSE
 	//the limit to the chemical and stat evolution
-	var/limited_borer = 10
+	var/limited_borer = 20
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:168]: oh thats a huge buff... we can always see how this goes.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 3]
```diff
@@ -164,11 +165,11 @@ GLOBAL_LIST_EMPTY(cortical_borers)
...
 	///borers can only enter biologicals if true
 	var/organic_restricted = TRUE
 	///borers are unable to enter changelings if true
-	var/changeling_restricted = TRUE
+	var/changeling_restricted = FALSE
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:172]: are you sure we want this? I vaguely remember people REALLY complaining about antag + antag.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 4]
```diff
@@ -77,9 +77,7 @@
 				return
 			cortical_owner.reagent_holder.reagents.add_reagent(reagent, cortical_owner.injection_rate_current, added_purity = 1)
 			cortical_owner.reagent_holder.reagents.trans_to(cortical_owner.human_host, cortical_owner.injection_rate_current, methods = INGEST)
-			var/obj/item/organ/internal/brain/victim_brain = cortical_owner.human_host.getorganslot(ORGAN_SLOT_BRAIN)
-			if(victim_brain)
-				cortical_owner.human_host.adjustOrganLoss(ORGAN_SLOT_BRAIN, 5)
+			cortical_owner.human_host.hurt_organ_randomly(rand(1,10))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:80]: incoming heart attack-- might be fine.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 5]
```diff
@@ -353,302 +345,10 @@
...
-		for(var/datum/quirk/quirky as anything in cortical_owner.human_host.quirks)
-			if(istype(quirky, /datum/quirk/item_quirk/allergic))
-				var/datum/quirk/item_quirk/allergic/allergies_quirk = quirky
-				var/allergies = allergies_quirk.allergy_string
-				render_list += "<span class='alert ml-1'>Subject is extremely allergic to the following chemicals:</span>\n"
-				render_list += "<span class='alert ml-2'>[allergies]</span>\n"
-
-		// we handled the last <br> so we don't need handholding
-		to_chat(cortical_owner, examine_block(jointext(render_list, "")), trailing_newline = FALSE, type = MESSAGE_TYPE_INFO)
+	healthscan(owner, cortical_owner.human_host, advanced = TRUE) // :thinking:
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:348]: thank you.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm [hunk 6]
```diff
@@ -105,11 +105,12 @@ GLOBAL_LIST_EMPTY(cortical_borers)
 										/datum/reagent/medicine/morphine,
 										/datum/reagent/medicine/inacusiate,
 										/datum/reagent/medicine/oculine,
+										/datum/reagent/medicine/mannitol,
```
Comment (Zonespace27) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:108]: why not neurine

Comment (nevimer) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:108]: I feel like the original balance of having mannitol but needing to learn neurine was fine

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 7]
```diff
@@ -973,23 +657,18 @@
...
-	if(host_choice != "Yes")
-		owner.balloon_alert(owner, "host not willing!")
-		StartCooldown()
+	if(cortical_owner.human_host.mind.has_antag_datum(/datum/antagonist/changeling))
+		to_chat(owner, "Host is already a changeling!")
 		return
+	cortical_owner.chemical_storage -= required_chems
+	if(do_after(cortical_owner, 15 SECONDS) && !cortical_owner.human_host.mind.has_antag_datum(/datum/antagonist/changeling))
+		GLOB.gifted_hosts += cortical_owner.human_host.ckey
+		cortical_owner.human_host.mind.add_antag_datum(/datum/antagonist/changeling)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:669]: oh, you're making someone into a ling... this is going to turn bad fast. Not strongly against it, but you really need some good justifications.

File: modular_skyrat/modules/cortical_borer/code/procs.dm [hunk 8]
```diff
@@ -0,0 +1,5 @@
+/mob/living/carbon/human/proc/hurt_organ_randomly(damage) // Gamble on your organs...
+	for(var/i in internal_organs)
+		if(prob(100 / length(internal_organs)))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cortical_borer/code/procs.dm:3]: why are you using prob here? use rand() or something.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm (hunk 7 above)
Comment (nevimer) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:669]: Going to turn them into baby lings in a future commit being its own antag type.

File: modular_skyrat/modules/cortical_borer/code/procs.dm (hunk 8 above)
Comment (nevimer) [reply modular_skyrat/modules/cortical_borer/code/procs.dm:3]: i failed algebra yet i try

**Comment by Stalkeros2** [comment]
I enjoy this. Borers are largely beneficial/cooperative (latter is more accurate) "antags", and making the host into a changeling, -which is basically some cool apex chad symbiotic organism afaik-, is a great climax that'll possibly result in more RP.
Oh and borers finally got some physical "teeth", -indirect too-, so it's even better.
But once again, as I've mentioned, they better be its own, possibly slightly stripped down variation of a changeling, having to rely on the borer's revival and chemicals instead of their typical resurrection-gland abilities; very likely having a singular objective of assisting with the borer spread.
Locking borers to their hosts after they get changeling'd would also be nice.

**Comment by jjpark-kb** [comment]
Okay, I'm going to have to pull the plug on the ling thing. I know the intentions are great, but I just know this is going to turn into a bomb that will get borers on the potential chopping block. The rest of the changes regarding random organ damage is fine, the addition of chems back etc. is fine, but the ling thing is just a bit too much.

**Review by jjpark-kb — CHANGES_REQUESTED** [review]
Another PR was merged that touched a bit of borers-- if this PR wants to still be merged, it will need to update to master as well as remove the bits I spoke about previously.

And yes, I do know this is a draft, but nonetheless.

**Comment by nevimer** [comment]
yea sure give me a day or two

**Comment by weesnaw-ops** [comment]
i like the idea of giving security a real, actual reason to kill borers, given that theyre boring and purely beneficial now

**Comment by jjpark-kb** [comment]
The PR is stale, and there was another PR that modified borers in a similar fashion.

**Comment by nevimer** [comment]
Lol let me finish it I only have time do things on weekends >:(

File: modular_skyrat/modules/cortical_borer/code/cortical_borer.dm (hunk 3 above)
Comment (tf-4) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer.dm:172]: I really do not want this.

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm (hunk 7 above)
Comment (tf-4) [reply modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm:669]: Really not interested in having them create lings at *all*.

**Comment by nevimer** [comment]
gonna atomize it

**Pull request closed without merging**